localstack-core/localstack/services/s3/notifications.py

```diff
@@ -421,14 +423,18 @@ def _get_event_payload(
             ),
         )
 
-        if ctx.key_version_id:
+        if ctx.key_version_id and ctx.key_version_id != "null":
             # object version if bucket is versioning-enabled, otherwise null
             record["s3"]["object"]["versionId"] = ctx.key_version_id
 
         event_type = ctx.event_type.lower()
         if any(e in event_type for e in ("created", "restore")):
-            record["s3"]["object"]["size"] = ctx.key_size
-            record["s3"]["object"]["eTag"] = ctx.key_etag
+            if "deletemarker" not in event_type:
+                record["s3"]["object"]["eTag"] = ctx.key_etag
+                record["s3"]["object"]["size"] = ctx.key_size
+            else:
+                record["s3"]["object"]["eTag"] = "d41d8cd98f00b204e9800998ecf8427e"
```

Comment:
Can we please add a comment in the code here why we added the this top most key: similar to what you have mentioned in the PR description?